modules/oadp-timeouts.adoc

```diff
@@ -0,0 +1,194 @@
+// Module included in the following assemblies:
+//
+// * backup_and_restore/application_backup_and_restore/troubleshooting.adoc
+
+:_content-type: CONCEPT
+[id="oadp-timeouts_{context}"]
+= OADP timeouts
+
+Extending a timeout allows complex or resource-intensive processes to complete successfully without premature termination, reducing the likelihood of errors, retries or failures. However, it is necessary to strike a balance when extending timeouts. Setting excessively long timeouts may hide underlying issues in the process. Therefore, careful consideration and monitoring are necessary to determine an appropriate timeout value that meets the needs of the process and the overall system performance.
+
+Below are various OADP timeouts, with instructions of how and when to implement these parameters:
+
+[id="Restic-timeout_{context}"]
+== Restic timeout
+
+.Description
+`timeout` defines the Restic timeout. The default value is 1h.
+
+.Implementation
+.. Use this parameter for Restic backups with large volumes.
+.. For Restic backups only. Use this parameter if backups are timing out with the following error:
++
+[source,terminal]
+----
+level=error msg="Error backing up item" backup=velero/monitoring error="timed out waiting for all PodVolumeBackups to complete"
+----
+
+.Procedure
+* Edit the values in the `spec.configuration.restic.timeout` block of the `DataProtectionApplication` CR manifest, as in the following example:
+[source,yaml]
+----
+apiVersion: oadp.openshift.io/v1alpha1
+kind: DataProtectionApplication
+metadata:
+ name: <dpa_name>
+spec:
+  configuration:
+    restic:
+      timeout: 1h
+  ...
+----
+
+[id="Velero-resource-timeout_{context}"]
+== Velereo resource timeout
+
+.Description
+`resourceTimeout` defines how long to wait for several Velero resources before timeout occurs, such as Velero CRD availability, `volumeSnapshot` deletion, and repo availability. The default is 10m.
+
+.Implementation
+.. Use this timeout for backups with very large volumes . This parameter is used as a timeout value when Velero tries to clean-up or delete the Container Storage Interface (CSI) snapshots, before marking the backup as complete.
```

Comment:
Global change > volumes/storage volumes

Reply:
Is there a hard metric to consider to determine a _very large volumes_.

Reply:
Good point @mburke5678 
Let's ask scale team, I asked them in the Jira ticket:
https://issues.redhat.com/browse/OADP-2041

Reply:
Very large - Large - total PV data usage >1TB